Add description for Jupyter to README.md

kloimhardt opened the pull request:
This is a suggestion for a usage description of Noj+Jupyter. It could be included in any other place than README.md

daslu commented:
@kloimhardt @behrica 

I am proposing the following -- what do you think?

* To solve the problem of the changing version, we can include the above text in a chapter of a Noj book rather than the README. In that chapter, we will fetch the current version string and print the code relevant to that version.

* We will add a very basic Getting Started chapter, that will include a few code examples that are currently compatible with the Clojupyter JAR. We will link to that chapter as a code example, rather than to the currently linked tutorial (which is currently not supported in Clojupyter).

kloimhardt commented:
@daslu Sounds good. This PR was just a proposal. I am not keen to actually getting it pulled. It can be closed at any time. But I leave it open as a reminder for now.

daslu commented:
Thanks. 
Maybe we can merge it as a first step, and then (if you wish), collaborate on turning it into a chapter in the book.

Let us see what @behrica thinks.

behrica commented:
looks good to me.
Maybe you can add additional a link to https://github.com/qubit55/clojupyter_colab_setup
as an alternative way to run this kernel in Google Colab ?

kloimhardt commented:
Added the Colab link and removed the link to the getting-started notebook. Feedback welcome. However, this text is free to use for anyone and any purpose deemed to fit, without any need to get an opinion from my side.

daslu commented:
@behrica I'm confused.

If I understand correctly, the Colab setup does not support Noj yet.
https://github.com/qubit55/clojupyter_colab_setup/issues/5

So, isn't it confusing to mention it hee?

behrica commented:
Yes, the download location is hardcoded.  So to be forked and changed. I
opened an issue to have it configurable. I can revise the text accordingly

Daniel Slutsky ***@***.***> schrieb am Sa., 8. Feb. 2025,
17:02:

> @behrica <https://github.com/behrica> I'm confused.
>
> If I understand correctly, the Colab setup does not support Noj yet.
> qubit55/clojupyter_colab_setup#5
> <https://github.com/qubit55/clojupyter_colab_setup/issues/5>
>
> So, isn't it confusing to mention it here?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/scicloj/noj/pull/132#issuecomment-2645807174>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAA7DANFML3JAGS3LH3YKQT2OYS7XAVCNFSM6AAAAABWWC3FFSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMNBVHAYDOMJXGQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

behrica commented:
Nevertheless it's worth to link to it . It's not obvious how to do this. Hopefully soon it's configurable .

daslu commented:
@behrica Thanks.

Would you like to merge this and then revise the text?

daslu commented:
I suggest adding a clarifying note that currently Jupyter supports most but not all data visualization kinds.

daslu commented:
Should we merge this?

kloimhardt commented:
I am closing this PR to clean the slate a bit. This PR is about a few lines of text only. Which are not lost and can be copied into any place desired (maybe README.md is not the right place).

daslu commented:
@behrica 
I think we can merge this.

daslu commented:
@behrica 
Would it be ok to merge this?

behrica commented:
Maybe we need to extend it know and add the Clay server start

daslu commented:
@behrica 
I think we can do that in a couple of days, after the Clay CLI is announced stable.
But it should not prevent us from merging the current PR. Right?

behrica commented:
ok, go ahead.

I think we still miss two aspects, but can do later:
1. "Noj" is a swell a normal clojure library (combining ....)
2. The clay server can be started as well via: clj -Sdeps '{:deps {org.scicloj/noj {:mvn/version "2-beta10.1"}}}' -M -m scicloj.clay.v2.main (when having clojure installed)

daslu commented:
@kloimhardt 
These days, I am going through various documentation tasks, and one of them is to merge this and add the details Carsten proposed. I am a bit slow, but will get there.

Whichever you prefer -- we can also close this if that is better.

Do you mean there is any detail that is wrong and should be fixed?

kloimhardt commented:
@daslu thank you, the PR can easily be left open for another month or so. The text of this PR is logically correct, today I made sure that the commands all work as expected.

daslu commented:
Thanks!
